### ISSUE TITLE
Cannot use recursive @ApiModelProperty

### ISSUE THREAD

**Issue by jlaffer** [nestjs/swagger#136]
## I'm submitting a...
<pre><code>
[ ] Regression 
[x] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
First - thanks for the AWESOME nestjs framework and addons!

Here's my problem:
I would like to use an `@ApiModelProperty` of `this` classes type.
example:
```typescript
export class FieldGroup {
  @ApiModelProperty({type: FieldGroup, isArray: true})
   subGroups?: Ref<FieldGroup>[];
}
```

This produces the following exception:
```
 UnhandledPromiseRejectionWarning: RangeError: Maximum call stack size exceeded
    at Array.find (native)
    at modelProperties.map.key (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:88:27)
    at Array.map (<anonymous>)
    at Object.exports.exploreModelDefinition (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:80:48)
    at modelProperties.map.key (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:90:45)
    at Array.map (<anonymous>)
    at Object.exports.exploreModelDefinition (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:80:48)
    at modelProperties.map.key (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:90:45)
    at Array.map (<anonymous>)
    at Object.exports.exploreModelDefinition (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:80:48)
    at modelProperties.map.key (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:90:45)
    at Array.map (<anonymous>)
    at Object.exports.exploreModelDefinition (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:80:48)
    at modelProperties.map.key (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:90:45)
    at Array.map (<anonymous>)
    at Object.exports.exploreModelDefinition (/myProjectPath/server/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:80:48)
```

## Environment

<pre><code>
Nest version: 5.3.7
@nestjs/swagger@2.5.1
 
For Tooling issues:
- Node version: v8.11.2  
- Platform:  Mac

</code></pre>

**Comment by riadhriadh** [comment]
export class FieldGroup {
  @ApiModelProperty({type: FieldGroup, isArray: true})
@Get('subGroups')
 subGroups?: Ref<FieldGroup>[];
}

**Comment by Epenance** [comment]
Also having this issue, its super unfortunate having to change the children of my object to an array of any for my application being able to build in CI :(

**Comment by aciccarello** [comment]
I think this was fixed with #179
